lib/amqp/gen/consumer.ex

```diff
@@ -183,25 +185,52 @@ defmodule Amqpx.Gen.Consumer do
 
   defp handle_message(
          message,
-         %{delivery_tag: tag, redelivered: redelivered} = meta,
+         %{delivery_tag: tag} = meta,
          %__MODULE__{
            handler_module: handler_module,
            handler_state: handler_state,
-           backoff: backoff
+           channel: channel
          } = state
        ) do
-    {:ok, handler_state} = handler_module.handle_message(message, meta, handler_state)
-    Basic.ack(state.channel, tag)
-    %{state | handler_state: handler_state}
+    case handler_module.handle_message(message, meta, handler_state) do
+      {:ok, handler_state} ->
+        Basic.ack(channel, tag)
+        %{state | handler_state: handler_state}
+
+      {:error, _} = error ->
+        Logger.error(inspect(error))
+        reject_message(meta, state)
+
+        state
+    end
   rescue
     e in _ ->
```

Comment:
già che ci siamo sistemerei anche la rescue, che così com'è non becca le exit a occhio

Reply:
Non farei reject on exit, perchè così il messaggio rimane e può essere riprocessato. Un esempio utile è quello del drain su gutexberg dove se durante il deploy ci arriva un SIGTERM allora usciamo e non facciamo nè ack nè nack del messaggio e quindi torna in coda.

Reply:
ma quella è una roba che ti arriva da fuori, così se uno ha configurato il reque e reque_after lo stai ignorando SOLO per le exit, ad esempio metti che nell'handler fai un Task.async se per caso si rompe il processo esce

Reply:
intendo che sono due casi distinti (il SIGTERM da fuori e un processo che esce nel mezzo di una funzione)

Reply:
ahhhh ho capito!